protocol/collectd/collectd.go

```diff
@@ -84,6 +84,47 @@ func addIfNotNullOrEmpty(dimensions map[string]string, key string, cond bool, va
 	}
 }
 
+// try to pull out dimensions out of name in the format name[k=v,f=x]-morename would
+// return name-morename and extract dimensions (k,v) and (f,x)
+// if we encounter something we don't expect use original
+func getDimensionsFromName(val *string) (instanceName string, toAddDims map[string]string) {
```

Comment:
You don't need to take val as a pointer.  Strings are themselves pointers to bytes similar to Java String.